components/fxa-client/src/http_client.rs

```diff
@@ -658,6 +688,14 @@ pub struct DeviceResponseCommon {
     pub push_endpoint_expired: bool,
 }
 
+#[derive(Deserialize, Debug)]
+pub struct AttachedClientsResponse {
```

Comment:
This should have all the fields from the response from the API. Then your calling code should do the filtering of responses and have a smaller return type.